### ISSUE TITLE
turn simplify_bool into normalise.

### ISSUE THREAD

**Issue by Wout4** [CPMpy/cpmpy#427]
There is more normalisation we should do before flattening, the natural place for this is in simplify bool.
The most pressing example is canonicalisation, which would make flattening faster in some cases, will help with cse, and removes the need to do it later in the transformation chain.

**Comment by IgnaceBleukx** [comment]
I agree, we should make a format what we consider "normalized" expressions.
Two things which would make sense to me are:
- Constraints taking a "collection" of arguments (alldiff, sum,...) should have their arguments sorted
- Convert numpy constants into Python constants. This will massively simplify our solver interfaces as solvers often do not accept numpy input

**Comment by OrestisLomis** [comment]
Some of the suggested improvements are already merged #996, #969, #1010, #959, ... However other things; mainly what @IgnaceBleukx mentioned still remains open.
